forbidigo/forbidigo.go

```diff
@@ -163,44 +195,198 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
 			return nil
 		}
 		return v
+	// The following two are handled below.
 	case *ast.SelectorExpr:
 	case *ast.Ident:
+	// Everything else isn't.
 	default:
 		return v
 	}
+
+	// The text as it appears in the source is always used because issues
+	// use that. It's used for matching unless usage of type information
+	// is enabled.
+	srcText := v.textFor(node)
+	matchText, pkgText := v.updateText(node, srcText)
 	for _, p := range v.linter.patterns {
-		if p.pattern.MatchString(v.textFor(node)) && !v.permit(node) {
+		if p.re.MatchString(matchText) &&
+			(p.Package == "" || p.pkgRe.MatchString(pkgText)) &&
+			!v.permit(node) {
 			v.issues = append(v.issues, UsedIssue{
-				identifier: v.textFor(node),
-				pattern:    p.pattern.String(),
+				identifier: srcText, // Always report the expression as it appears in the source code.
+				pattern:    p.re.String(),
 				pos:        node.Pos(),
-				position:   v.fset.Position(node.Pos()),
-				customMsg:  p.msg,
+				position:   v.runConfig.Fset.Position(node.Pos()),
+				customMsg:  p.Msg,
 			})
 		}
 	}
 	return nil
 }
 
+// textFor returns the expression as it appears in the source code (for
+// example, <importname>.<function name>).
 func (v *visitor) textFor(node ast.Node) string {
 	buf := new(bytes.Buffer)
-	if err := printer.Fprint(buf, v.fset, node); err != nil {
-		log.Fatalf("ERROR: unable to print node at %s: %s", v.fset.Position(node.Pos()), err)
+	if err := printer.Fprint(buf, v.runConfig.Fset, node); err != nil {
+		log.Fatalf("ERROR: unable to print node at %s: %s", v.runConfig.Fset.Position(node.Pos()), err)
 	}
 	return buf.String()
 }
 
+// pkgTextFor expands the selector in a selector expression to the full package
```

Comment:
The comment name here doesn't match the function.  A name that describes its function might be welcome, like `expandMatchText`.

Reply:
After checking the current state again I see what you mean: a literal mismatch between the comment and the function name.

`expandMatchText` works for me.

Reply:
Done.